parachain/test-parachains/adder/collator/Cargo.toml

```diff
@@ -28,6 +32,11 @@ sp-core = { git = "https://github.com/paritytech/substrate", branch = "master" }
 sc-authority-discovery = { git = "https://github.com/paritytech/substrate", branch = "master" }
 sc-service = { git = "https://github.com/paritytech/substrate", branch = "master" }
 
+# This one is tricky. Even though it is not used directly by the collator, we still need it for the
+# `puppet_worker` binary, which is required for the integration test. However, this shouldn't be
+# a big problem since it is used transitively, at least when built with `real-overseer`.
+polkadot-node-core-pvf = { path = "../../../../node/core/pvf", optional = true }
```

Comment:
Yeah, each time we need to create a puppet worker. The alternative where you want to test PVF execution is to avoid using the whole validation host with external processes and all that jazz, but that would require you to depend on `polkadot-node-core-pvf` as well. That would be a dev-dependency though.